### PR TITLE
Abuse-monitor: Sort recommendations by user

### PR DESCRIPTION
**Pull request by mia-pi-git**
Convenience for staff members, so they can see who got which recommendation. Mostly a PR since this involves a migration.

### CODE REVIEW

File: server/chat-plugins/abuse-monitor.ts [hunk 1]
```diff
@@ -42,16 +42,31 @@ const REPORT_NAMECOLORS: {[k: string]: string} = {
...
+} = (() => {
+	const plugin = global.Chat?.oldPlugins['abuse-monitor'];
+	if (!plugin?.cache) return {};
+	if (plugin.migrated) return plugin.cache;
+	for (const k in plugin.cache) {
+		const cur = plugin.cache[k];
+		if (typeof cur.recommended?.type === 'string') { // would be object if it was the new entry
+			// we cannot feasibly determine who it was (but it __is__ logged in <<abuselog>>, so staff can)
+			delete cur.recommended;
+		}
```
Comment (AnnikaCodes) [inline server/chat-plugins/abuse-monitor.ts:65]: Should we really discard data about recommendations just because we don't know which user it was for?

Comment (mia-pi-git) [reply server/chat-plugins/abuse-monitor.ts:65]: Not like there's a better way to do it. Plus, it's logged to abuselog anyway, so staff can still see it.

Comment (AnnikaCodes) [reply server/chat-plugins/abuse-monitor.ts:65]: You could save it under something like `an unknown user`, but it's probably fine as-is.

**Comment by AnnikaCodes** [comment]
Thanks!

**Pull request merged**